utils/controller.js

```diff
@@ -23,4 +25,42 @@ exports.insertProduct = function(body, barcode, res) {
     console.log(`stored ${data} in mongodb`);
     return res.status(StatusCode.CREATED).send(newProduct);
   });
+}
+
+exports.getSimilarCategory = function(category, res) {
+  connectDb().then(async () => {
+    try {
+      let docs = await models.Product.aggregate([
+        {$match: {
+          'category': category,
+          "ESG":{$ne:null}
+        }},
+        {$group: {
+          _id: '$manufacturer',
+          manufacturer: { "$first" : "$manufacturer"},
+          ESG: { "$first": "$ESG" }
+        }},
+        { $sort: { "ESG": -1 } },
+        { $limit: 5 },
+        { $project : {
+          _id : 0
+        }}
+      ]);
+      if(docs) {
+        console.log(docs);
+        return res.status(StatusCode.OK).send({docs});
+      }
+
+      if(!docs) {
```

Comment:
This can be simplified to:

`return res.status(StatusCode.BAD_REQUEST).send();`

You don't need to check if the document exists again.

Reply:
Keep this:
```
exports.getSimilarCategory = function(category, res) {
  connectDb().then(async () => {
    try {
      let docs = await models.Product.aggregate([
        {$match: {
          'category': category,
          "ESG":{$ne:null}
        }},
        {$group: {
          _id: '$manufacturer',
          manufacturer: { "$first" : "$manufacturer"},
          ESG: { "$first": "$ESG" }
        }},
        { $sort: { "ESG": -1 } },
        { $limit: 5 },
        { $project : {
          _id : 0
        }}
      ]);
      if(docs) {
        console.log(docs);
        return res.status(StatusCode.OK).send({docs});
      }
```
Change this:
```suggestion
     // else
        return res.status(StatusCode.BAD_REQUEST).send();
```
Keep this:
```      
    } catch(err) {
      console.error(err);
      return res.status(StatusCode.BAD_REQUEST).send(err);
    }

  }).catch((err) => {
    console.error(err);
    return res.status(StatusCode.INTERNAL_SERVER_ERROR).send(err);
  });
}
```